www/app.py

```diff
@@ -75,70 +74,46 @@ class HeartBeatManager():
+    def __init__(self, client_id, client_secret, id=None):
         """Constructor for the HeartBeatManager."""
         self.run = True
         self.thread = None
         self.web_id = id
-        self.payload = payload
-        self.session = requests.session()
-
-        # Log in to rovercode-web
-        self.session.get(ROVERCODE_WEB_LOGIN_URL + '/')
-        try:
-            csrftoken = self.session.cookies['csrftoken']
-        except KeyError:
-            # if the server doesn't provide a csrftoken, no problem
-            csrftoken = ''
+        self.client_id = client_id
+        self.name = None
+        self.local_ip = get_local_ip()
 
+        # Log in to rovercode-web using oauth credentials
         login_data = {
-            'login':user_name,
-            'password':user_pass,
-            'csrfmiddlewaretoken':csrftoken,
+            'grant_type':'client_credentials',
+            'client_id':client_id,
+            'client_secret':client_secret,
         }
-        self.session.post(ROVERCODE_WEB_LOGIN_URL + '/', data=login_data)
-
-        # We have a new csrf token after logging in
-        try:
-            csrftoken = self.session.cookies['csrftoken']
-        except KeyError:
-            # if the server doesn't provide a csrftoken, no problem
-            csrftoken = ''
-        self.payload['csrfmiddlewaretoken'] = csrftoken
-        self.csrftoken = csrftoken
-
-    def create(self):
-        """Register the rover initially with rovercode-web."""
-        r = self.session.post(ROVERCODE_WEB_REG_URL + '/', self.payload)
-        info = json.loads(r.text)
-        try:
-            self.web_id = info['id']
-            init_input_gpio(info['left_eye_pin'], info['right_eye_pin'])
-            print "rovercode-web id is " + str(self.web_id)
-        except KeyError:
-            self.web_id = None
-        return r
+        r = requests.post(ROVERCODE_WEB_OAUTH2_URL + '/', data=login_data)
+        print r.text
+        self.access_token = r.json()['access_token']
+        print "Access token is: " + self.access_token
 
     def update(self):
         """Check in with rovercode-web, updating our timestamp."""
-        headers = {'X-CSRFTOKEN':self.csrftoken}
-        return self.session.put(ROVERCODE_WEB_REG_URL+"/"+str(self.web_id)+"/",
-                                data = self.payload,
-                                headers = headers)
+        headers = {'Authorization':'Bearer '+self.access_token}
+        payload = {'name': self.name, 'local_ip': self.local_ip}
+        return requests.put(ROVERCODE_WEB_REG_URL+"/"+str(self.web_id)+"/",
+                                data=payload,
+                                headers=headers)
 
     def read(self):
         """Look for our name on rovercode-web. Sets web_id if found."""
-        global rover_name
-        result = self.session.get(ROVERCODE_WEB_REG_URL+'?name='+rover_name)
+        headers = {'Authorization':'Bearer '+self.access_token}
```

Comment:
Since the headers need to be created for everything, I would suggest creating some kind of method like `web_api_get` to wrap `requests.get` that fills that in.

Reply:
Probably the best answer would be to make a client class that you could call `get()`, `post()`, etc on that would handle the authentication within. We could even make it a standalone project that this project imports (and anyone else who would want to interact with our API). But, for now, this is a good option.

Reply:
Oh, that's a really good idea. Made a ticket: https://github.com/rovercode/rovercode/issues/128